syntax = "proto3";

option go_package = "github.com/e-conomic/hiring-assignments/machinelearningteam/summary-statistics-service/proto";

package statistics;

service DocumentSummarizer {
  rpc SummarizeDocument (SummarizeDocumentRequest) returns (SummarizeDocumentReply) {}
}

message SummarizeDocumentRequest {
  Document document = 1;
}

message SummarizeDocumentReply {
  bytes content = 1;
}

message Document {
  // Document content, represented as a stream of bytes.
  // Note: As with all `bytes` fields, protobuffers use a pure binary
  // representation, whereas JSON representations use base64.
  bytes content = 1;
  // If both content and source is present use content.
  DocumentSource source = 2;
}

message DocumentSource {
  // The URI of the source document. Can be either:
  //
  // 1. A publicly-accessible image HTTP/HTTPS URL.
  string http_uri = 1;
}

// Froblems defining this in separate file has to life here for the moment.
service DocumentSummarizerBackend {
  rpc SummarizeDocument (SummarizeDocumentWorkload) returns (statistics.SummarizeDocumentReply) {}
}

message SummarizeDocumentWorkload {
  // workload for the backend service
  statistics.SummarizeDocumentRequest request = 1;
  repeated string aggregateColumns = 2;
  repeated string excludedColumns = 3;
}

